kong/client.go

```diff
@@ -164,6 +165,34 @@ func NewClient(baseURL *string, client *http.Client) (*Client, error) {
 	return kong, nil
 }
 
+// WithWorkspace return the client with a defined workspace
+func (c *Client) WithWorkspace(workspace string) *Client {
+	c.SetWorkspace(workspace)
+	return c
+}
+
+// SetWorkspace defines a workspace for the client
+func (c *Client) SetWorkspace(workspace string) {
+	c.workspace = workspace
+}
```

Comment:
```suggestion
// SetWorkspace defines a workspace for the client
func (c *Client) SetWorkspace(workspace string) {
	c.workspaceLock.Lock()
	defer c.workspaceLock.Unlock()
	c.workspace = workspace
}
```

Also please implement `GetWorkspace` using `RLock()` and `RUnlock()` appropriately.